servers/curated-corpus-api/src/admin/resolvers/mutations/Section/index.ts

```diff
@@ -134,6 +138,67 @@ export async function createCustomSection(
   return await dbCreateCustomSection(context.db, data);
 }
 
+/**
+ * Updates an existing custom editorial section.
+ * 
+ * This mutation allows curators to modify sections created with MANUAL source.
+ * It performs comprehensive validation including:
+ * - Existence check for the section
+ * - Source type validation (must be MANUAL)
+ * - Permission validation for both current and target surfaces
+ * - IAB metadata validation if provided
+ * 
+ * @param parent - GraphQL parent resolver
+ * @param data - UpdateCustomSectionInput containing section updates
+ * @param context - Admin context with auth and database access
+ * @returns Updated Section with associated SectionItems
+ * @throws UserInputError - If section not found, invalid source, or validation fails
+ * @throws AuthenticationError - If user lacks required permissions
+ */
+export async function updateCustomSection(
+  parent,
+  { data },
+  context: IAdminContext,
+
+): Promise<Section> {
+  const { externalId } = data;
+
+  // Find the existing section
+  const existingSection = await context.db.section.findUnique({
+    where: { externalId },
+  });
+
+  if (!existingSection) {
+    throw new NotFoundError(`Cannot update section: Section with id "${externalId}" does not exist.`);
+  }
+
+  // Check if the existing section is not a custom section
+  if (existingSection.createSource !== ActivitySource.MANUAL) {
+    throw new UserInputError(
+      `Section with externalId ${externalId} is not a custom (MANUAL) Section and cannot be updated using this mutation`,
+    );
+  }
+
+  // Check if the user can perform this mutation
+  if (!context.authenticatedUser.canWriteToCorpus()) {
```

Comment:
i'm not sure of the exact requirements here, but should we be checking if the authenticated user can write to the surface this section is assigned to? we do this in other places, e.g. [creating scheduled items](https://github.com/Pocket/content-monorepo/blob/main/servers/curated-corpus-api/src/admin/resolvers/mutations/ScheduledItem/index.ts#L151).

Reply:
@jpetto good question! 
It didn't seem right to pass the surface ID when updating a section -- I think the reason we did that for ML section updates was because the create / update mutation is shared.

Reply:
@jwalterss , lets replace this check with: 
```
  // Check if the user can execute this mutation.
  if (!context.authenticatedUser.canWriteToSurface(existingSection.scheduledSurfaceGuid)) {
    throw new AuthenticationError(ACCESS_DENIED_ERROR);
  }
```